### DISCUSSION TITLE
SocketNotConnected or ConnectionAborted

### DISCUSSION THREAD

**Discussion by mike9kk** [cefsharp/CefSharp#3926] [General]
My project doesn't load pages anymore just getting blank sites. So I switched to minimalExample with CefSharp.MinimalExample.netcore.sln to figure out the issue. After a couple startups it has the same problems with following errors starting to appear more often instead of the page:
`Failed to load URL http://www.google.com/ with error ERR_SOCKET_NOT_CONNECTED (SocketNotConnected).`
`Failed to load URL http://www.google.com/ with error ERR_CONNECTION_ABORTED (ConnectionAborted).`
This is with unchanged code. The google page sometimes still shows but other pages pretty much stop to work at that point (when entering them in the address bar).
I tried to delete the cache folder from the CefSettings CachePath but it didn't help. Are there any other data saved by CefSharp? As it looks like things got worse the more often it's run.

**Comment by amaitland** [comment]
Have you tried on a different computer? What antivirus/firewall are you using? 

Sounds like something specific to your network/environment.

**Reply by mike9kk** [reply]
It looks like on a different computer it works so far. But the computer with the issues did also run for months fine before the issue started to appear. 
Doesn't look like it's the antivirus/firewall as they are the same for both systems even same windows and network. Disabling them also didn't help. Every other software I'm using runs fine on the computer with the issue even the normal chrome.

**Reply by amaitland** [reply]
> Every other software I'm using runs fine on the computer with the issue even the normal chrome

Does Chrome run correctly? 

I'd be running system file check. 

https://support.microsoft.com/en-us/topic/use-the-system-file-checker-tool-to-repair-missing-or-corrupted-system-files-79aa86cb-ca52-166a-92a3-966e85d4094e

**Reply by mike9kk** [reply]
Yes the normal Chrome runs fine. I can browse, watch videos or download with it at the side while CefSharp produces the errors from above.

Dism and sfc says it didn't find any integrity violations.

**Reply by amaitland** [reply]
What version of Chrome? What version of CefSharp? The minimum example was updated to version 96 yesterday.

**Reply by mike9kk** [reply]
Chrome is 96.0.4664.93
The CefSharp error happens with both before the update and with the update from yesterday. 

I'm now able to reproduce the issue better after starting to ignore the initial google page as that one sometimes loads (possible from cache or it uses some special method?) at a point where other pages already run into the issue.
The following steps reproduce the issue reliable on my system:

1. extract minimal example zip
2. open CefSharp.MinimalExample.netcore.sln
3. right click solution->build or rebuild (doesn't matter)
4. run
5. at this point it always works not producing any errors!
6. close it
7. run again
8. now it might show the error or still work, repeating 6-7 will break it after at most 3 runs
9. after it broke once any non google site won't work anymore and wont be fixed by running it again, rebuilding or closing and reopening visual studio

Something else I noticed: after closing visual studio the folder with the project can't be deleted for about 1 minute (complains about files being in use). Which is unusually long.
But after waiting for that ~minute and renaming the folder before opening the project will put one back to step 4 where it will work at least the first time but break again doing 6-8.

Reproduceable only on one of the two tested systems.

**Reply by amaitland** [reply]
> But the computer with the issues did also run for months fine before the issue started to appear.

What changed with this computer? I've not seen any other reports of the same problem and everything works as expected for me. At this point in time it looks like a problem with your specific machine.

> Are there any other data saved by CefSharp? 

CachePath and [UserDataPath](https://cefsharp.github.io/api/96.0.x/html/P_CefSharp_CefSettingsBase_UserDataPath.htm)

> The following steps reproduce the issue reliable on my system:

Using the `WinForms` example, no issues.

> Something else I noticed: after closing visual studio the folder with the project can't be deleted for about 1 minute (complains about files being in use). Which is unusually long.

You'd need to check which process has the files in use for this to be meaningful information.

**Reply by mike9kk** [reply]
> CachePath and [UserDataPath](https://cefsharp.github.io/api/96.0.x/html/P_CefSharp_CefSettingsBase_UserDataPath.htm)

Ok thanks deleting them didn't change it.

> You'd need to check which process has the files in use for this to be meaningful information.

It was PerfWatch2 blocking the files. Disabling it in studio allows now to rename the folder directly but didn't change the issue.

> What changed with this computer? I've not seen any other reports of the same problem and everything works as expected for me. At this point in time it looks like a problem with your specific machine.

There was a break without testing of about a month between the last time when it was working without issues and the issue appearing the first time. The windows updates were installed but only the small ones they release regularly with no big changes otherwise I'm remembering.

Thanks for the help so far. It appears to be a strange issue and like you mentioned no reports from others. I will switch for now to my other system where it still works. As long as it's only with one computer I can work around it.
